app/src/main/java/com/infomaniak/mail/ui/main/folder/TwoPaneViewModel.kt

```diff
@@ -0,0 +1,130 @@
+import com.infomaniak.mail.ui.newMessage.NewMessageActivityArgs
+import com.infomaniak.mail.utils.coroutineContext
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import javax.inject.Inject
+
+@HiltViewModel
+class TwoPaneViewModel @Inject constructor(
+    private val draftController: DraftController,
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
+) : ViewModel() {
+
+    private val ioCoroutineContext = viewModelScope.coroutineContext(ioDispatcher)
+
+    val currentThreadUid = MutableLiveData<String?>(null)
+
+    inline val isInThread get() = currentThreadUid.value != null
+    val isInSearch = MutableLiveData(false)
+    val rightPaneFolderName = MutableLiveData<String?>(null)
+    var previousFolderId: String? = null
+
+    val downloadAttachmentArgs = SingleLiveEvent<DownloadAttachmentProgressDialogArgs>()
+    val newMessageArgs = SingleLiveEvent<NewMessageActivityArgs>()
+    val replyBottomSheetArgs = SingleLiveEvent<ReplyBottomSheetDialogArgs>()
+    val threadActionsArgs = SingleLiveEvent<ThreadActionsBottomSheetDialogArgs>()
+    val messageActionsArgs = SingleLiveEvent<MessageActionsBottomSheetDialogArgs>()
+    val detailedContactArgs = SingleLiveEvent<DetailedContactBottomSheetDialogArgs>()
+
+    fun openThread(uid: String) {
+        currentThreadUid.value = uid
+    }
+
+    fun closeThread() {
+        currentThreadUid.value = null
+    }
+
+    fun navigateToSelectedDraft(message: Message) = liveData(ioCoroutineContext) {
+        emit(
+            NewMessageActivityArgs(
+                arrivedFromExistingDraft = true,
+                draftLocalUuid = draftController.getDraftByMessageUid(message.uid)?.localUuid,
+                draftResource = message.draftResource,
+                messageUid = message.uid,
+            ),
+        )
+    }
+
+    fun navigateToDownloadAttachment(resource: String, name: String, type: Attachment.AttachmentType) {
+        downloadAttachmentArgs.value = DownloadAttachmentProgressDialogArgs(resource, name, type)
+    }
+
+    fun navigateToNewMessage(
+        draftMode: DraftMode = DraftMode.NEW_MAIL,
+        previousMessageUid: String? = null,
+        shouldLoadDistantResources: Boolean = false,
+        arrivedFromExistingDraft: Boolean = false,
+        draftLocalUuid: String? = null,
+        draftResource: String? = null,
+        messageUid: String? = null,
+        mailToUri: Uri? = null,
+    ) {
+        newMessageArgs.value = NewMessageActivityArgs(
+            draftMode = draftMode,
+            previousMessageUid = previousMessageUid,
+            shouldLoadDistantResources = shouldLoadDistantResources,
+            arrivedFromExistingDraft = arrivedFromExistingDraft,
+            draftLocalUuid = draftLocalUuid,
+            draftResource = draftResource,
+            messageUid = messageUid,
+            mailToUri = mailToUri,
+        )
```

Comment:
parameters' names aren't really useful here

Reply:
Try to remove them, then come back to me :D